pkg/autopilot/controller/signal/k0s/cordon.go

```diff
@@ -147,9 +147,11 @@ func (r *cordoning) moveToNextState(ctx context.Context, signalNode crcli.Object
 // draining ignores daemonsets
 func (r *cordoning) drainNode(ctx context.Context, signalNode crcli.Object) error {
 	logger := r.log.WithField("signalnode", signalNode.GetName()).WithField("phase", "drain")
-	node, ok := signalNode.(*corev1.Node)
-	if !ok {
-		return fmt.Errorf("failed to covert signalNode to Node")
+
+	//get node from client
+	node := &corev1.Node{}
+	if err := r.client.Get(ctx, crcli.ObjectKey{Name: signalNode.GetName()}, node); err != nil {
```

Comment:
I think the Node lookup is only necessary when the object is not already a Node. So maybe shove this into the else branch?

Reply:
I think I understood what you meant. there wasn't an existing else branch in that codepath though